chaoslib/litmus/spring-boot-chaos/lib/spring-boot-chaos.go

```
@@ -0,0 +1,353 @@
+	})
+	log.InfoWithValues("[Info]: Chaos monkeys assaults will be injected to the target pods as follows", logrus.Fields{
+		"CPU Assault":       experimentsDetails.ChaosMonkeyAssault.CPUActive,
+		"Memory Assault":    experimentsDetails.ChaosMonkeyAssault.MemoryActive,
+		"Kill App Assault":  experimentsDetails.ChaosMonkeyAssault.KillApplicationActive,
+		"Latency Assault":   experimentsDetails.ChaosMonkeyAssault.LatencyActive,
+		"Exception Assault": experimentsDetails.ChaosMonkeyAssault.ExceptionsActive,
+	})
+
+	switch strings.ToLower(experimentsDetails.Sequence) {
+	case "serial":
+		if err := injectChaosInSerialMode(experimentsDetails, clients, chaosDetails, eventsDetails, resultDetails); err != nil {
+			return err
+		}
+	case "parallel":
+		if err := injectChaosInParallelMode(experimentsDetails, clients, chaosDetails, eventsDetails, resultDetails); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("%v sequence is not supported", experimentsDetails.Sequence)
+	}
+
+	// Waiting for the ramp time after chaos injection
+	if experimentsDetails.RampTime != 0 {
+		log.Infof("[Ramp]: Waiting for the %vs ramp time after injecting chaos", experimentsDetails.RampTime)
+		common.WaitForDuration(experimentsDetails.RampTime)
+	}
+	return nil
+}
+
+// CheckChaosMonkey verifies if chaos monkey for spring boot is available in the selected pods
+func CheckChaosMonkey(chaosMonkeyPort string, chaosMonkeyPath string, targetPods corev1.PodList) (bool, error) {
+	hasErrors := false
+
+	for _, pod := range targetPods.Items {
+		endpoint := "http://" + pod.Status.PodIP + ":" + chaosMonkeyPort + chaosMonkeyPath
+		log.Infof("[Check]: Checking pod: %v (endpoint: %v)", pod.Name, endpoint)
+
+		resp, err := http.Get(endpoint)
+		if err != nil {
+			log.Errorf("failed to request chaos monkey endpoint on pod %v (err: %v)", pod.Name, resp.StatusCode)
+			hasErrors = true
+		}
+
+		if resp.StatusCode != 200 {
```

In case of error `Get` will return `nil` for `resp`, so this statement will throw a runtime error. You can return the error directly rather than flagging it. Or you can use the `hasErrors` flag in this statement as well.

---

Damn you're right !
I will go to the `continue` option.